theseus/embodied/kinematics/robot_model.py

```diff
@@ -4,18 +4,22 @@
 # LICENSE file in the root directory of this source tree.
 
 import abc
+from typing import Dict, Union
 
 import torch
 
 from theseus.geometry import SE2, LieGroup, Point2, Vector
 
+RobotModelInput = Union[torch.Tensor, Vector]
+RobotModelOuput = Union[LieGroup, Dict[str, LieGroup]]
```

Comment:
I think `RobotModelInput` makes  sense as torch tensors and theseus vectors are effectively two representations of the same concept.
`RobotModelOutput` can be trouble some as the user can be unsure what to expect. I'm thinking maybe we can have two different FK implementations, a `forward_kinematics_to_link(RobotModelInput, str) -> LieGroup` that computes pose of a specified link, and a `forward_kinematics_all_links(RobotModelInput) -> Dict[str, LieGroup]` that computes poses of all links in the kinematics tree.
Do you think that makes sense? Would be great to have @mhmukadam 's input on this.

Reply:
> as the user can be unsure what to expect

Good point! Although the long function names would be tedious to use too. Here are two options:
1. We just use `Union[LieGroup, Dict[str, LieGroup]]` and not worry about `RobotModelOuput `
2. We have a wrapper function `forward_kinematics()` that internally calls either `..._to_link()` or `..._all_links()`

Reply:
I don't know what robot models outside of URDFs might be implemented in Theseus in the future, but I would expect most forward kinematics computations to be specified by an URDF file, thus having named links.
I'm actually thinking that if there isn't a second use case, maybe we shouldn't over-modularize things and maintain only one RobotModel class?

Reply:
@mhmukadam Any thoughts? I don't have a too strong opinion on this, but I'm hesitant about having a single `RobotModel` class that's hard wired to a specific format (URDF in this case). 

How about keeping the current class hierarchy but just have the output be a plain `Dict[str, LieGroup]` and revisit this later if needed?

Reply:
>  just have the output be a plain `Dict[str, LieGroup]` and revisit this later if needed?

This sounds good, and the hierarchy in Austin's PR: `KinematicsModel` -> `UrdfRobotModel`
I imagine human models, hand models, etc can later derive from `KinematicsModel`.